lib/modules/manager/gradle/extract.ts

```diff
@@ -70,6 +80,15 @@ export async function extractAllPackageFiles(
       } else if (isTOMLFile(packageFile)) {
         const updatesFromCatalog = parseCatalog(packageFile, content);
         extractedDeps.push(...updatesFromCatalog);
+      } else if (isGcvPropsFile(packageFile)) {
+        if (usesConsistentVersionPlugin && packageFile === VERSIONS_PROPS) {
+          // We use gradle-consistent-versions, and pass in both the versions file and the lock file
+          const updatesFromGcv = parseGcv(
+            fileContents[VERSIONS_PROPS]!,
+            fileContents[VERSIONS_LOCK]!
```

Comment:
This won't work for a repository with two distinct Gradle projects in separate subfolders. Probably have a look at `fs.getSiblingFileName()` on how to find props and lock files relative to `packageFile`.

Abstraction-wise I'd suggest you do `parseGcv(packageFile, fileContents)` and look up the lock file in there. Also the check if gradle-consistent-versions plugin occurs in a gradle file, should go there as it's nothing that's relevant for the extract step as a whole.

Nitpick: Please remove the comment, it's not needed here IMHO.

Reply:
So I reworked the whole detection and extraction logic to support monorepos, added test for it.